### PR TITLE
refactor: add redis dependency and configuration

### PR DESCRIPTION
**Pull request by GuillaumeTech**
Needs a model server to be usable

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -110,6 +110,28 @@ If you're using Botfront in a production environment, consider adding your own d
...
+### Enable and configure redis
+
+Redis is a data structure store, used by rasa as a lockstore. It should be enabled when working with multiple rasa instances. As it ensure that
+incoming messages are processed in the right order.
+
+| Parameter                            | Description                                                   | Default                             |
+|--------------------------------------|---------------------------------------------------------------|-------------------------------------|
+| `redis.enabled`                      | Set to `true` to enable redis                                 | `false`                             |
+| `redis.cluster.enabled`              | Do not change this value                                      | `false`                             |
+| `redis.cluster.slaveCount`           | Do not change this value                                      | 0                                   |
```
Comment (znat) [inline README.md]: Why exposing them if they cannot be changed?

File: README.md [hunk 2]
```diff
@@ -110,6 +110,28 @@ If you're using Botfront in a production environment, consider adding your own d
...
+| `redis.usePassword`                  | Enable password authentication for connecting to redis        | `true`                              |
+| `redis.global.password`              | The password value, if not set will be randomly generated     | `nil`                               |
+
+
+**Important: about .cluster.enabled and .cluster.slaveCount** 
+
+"Redis Cluster is not able to guarantee strong consistency. In practical terms this means that under certain conditions it is possible that Redis Cluster will lose writes that were acknowledged by the system to the client."
+So you should not use a redis cluster because rasa uses it for storing locks, thus it need to be consistent.
```
Comment (znat) [inline README.md]: it need**s**

File: botfront-project/templates/ingress.yaml [hunk 3]
```diff
@@ -9,6 +9,9 @@ metadata:
   name: {{ .Release.Name }}-rasa-ingress
   annotations:
     {{- if .Values.ingress.nginx.enabled}}
+    #uncomment the following line if you are working with multiple rasa using a websocket based channel
+    # web socket connection need to have a sticky session https://socket.io/docs/using-multiple-nodes/
+    #nginx.ingress.kubernetes.io/affinity: "cookie"
```
Comment (znat) [inline botfront-project/templates/ingress.yaml]: I think this should be a  `ingress.nginx.enableSessionAffinity`parameter enabled by default. The line should be surrounded by conditionals so it is only added if enabled. 
The documentation should mention that this should be enabled if a channel depending on websockets or long polling (such as Rasa Webchat) is used.

File: README.md [hunk 4]
```diff
@@ -110,6 +110,28 @@ If you're using Botfront in a production environment, consider adding your own d
 **Important**
 If you are using the provided MongoDB deployment, `mongodb.mongodbHost` must follow the following pattern: `<release-name>-mongodb-service.<namespace>`
 
+
+### Enable and configure redis
+
+Redis is a data structure store, used by rasa as a lockstore. It should be enabled when working with multiple rasa instances. As it ensure that
```
Comment (znat) [inline README.md]: We should also give an example of the lock store config to paragraph to add to endpoints to make it work.

File: botfront-project/templates/ingress.yaml [hunk 5]
```diff
@@ -9,6 +9,9 @@ metadata:
...
     kubernetes.io/ingress.class: nginx
     nginx.org/websocket-services: {{ .Release.Name }}-rasa-service
```
Comment (znat) [inline botfront-project/templates/ingress.yaml:17]: Same as above. `ingress.nginx.socketEnabled` (default: true).
Same: doc (or table) should be mention that it should be enabled if a channel using websockets or long polling (such as Rasa Webchat) is used.

File: README.md (hunk 1 above)
Comment (znat) [reply README.md]: Or you can expose it but leave the doc in the values.yaml

File: botfront/values.yaml [hunk 6]
```diff
@@ -124,4 +124,18 @@ mongo-express:
...
-    tls: []
\ No newline at end of file
+    tls: []
+
+redis: 
+  # Set enabled: true to deploy redis
+  enabled: false
+  # redis is used as a lock database for rasa, so async update between master and slaves can create issues
+  cluster:
+    enabled: false
```
Comment (znat) [inline botfront/values.yaml]: This probably where the risks of changing them should be explained (rather than the readme)

**Review by znat — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (GuillaumeTech) [reply README.md]: I removed those from the docs and added the warning in the values.yaml

File: README.md (hunk 4 above)
Comment (GuillaumeTech) [reply README.md]: The docs have been updated, now they describe bit more the lockstore and the sticky session mechanism

File: botfront-project/templates/ingress.yaml (hunk 3 above)
Comment (GuillaumeTech) [reply botfront-project/templates/ingress.yaml]: Move it to the table for rasa values

File: botfront-project/templates/ingress.yaml (hunk 5 above)
Comment (GuillaumeTech) [reply botfront-project/templates/ingress.yaml:17]: Move it to the table for rasa values and removed the old comment

**Pull request merged**